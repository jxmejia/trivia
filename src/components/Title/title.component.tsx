import { memo } from "react";
import { TitleProps } from "./title.definition";
import { StyledTitle } from "./title.style";

const Title = ({ children }: TitleProps): JSX.Element => {
  return <StyledTitle>{children}</StyledTitle>;
};

export default memo(Title);
