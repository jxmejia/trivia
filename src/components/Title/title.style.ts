import styled from "@emotion/styled";

export const StyledTitle = styled.div`
  font-weight: bold;
  font-size: 18px;
`;
