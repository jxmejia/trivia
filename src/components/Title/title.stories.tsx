import { ComponentStory, ComponentMeta } from "@storybook/react";
import Title from "./title.component";

export default {
  title: "Title",
  component: Title,
} as ComponentMeta<typeof Title>;

const Template: ComponentStory<typeof Title> = (args) => <Title {...args} />;

export const Default = Template.bind({});

Default.args = {
  children: "Trivia challenge",
};
