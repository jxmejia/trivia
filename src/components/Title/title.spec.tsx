import { createSerializer } from "@emotion/jest";
import { render } from "@testing-library/react";
import Title from "./title.component";

expect.addSnapshotSerializer(createSerializer());

describe("Title", () => {
  it("renders component", () => {
    const { container } = render(<Title />);
    expect(container).toMatchSnapshot();
  });
});
