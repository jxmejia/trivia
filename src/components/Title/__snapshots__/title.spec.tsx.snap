// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title is renders component 1`] = `
.emotion-0 {
  font-weight: bold;
  font-size: 18px;
}

<div>
  <div
    class="emotion-0"
  />
</div>
`;

exports[`Title renders component 1`] = `
.emotion-0 {
  font-weight: bold;
  font-size: 18px;
}

<div>
  <div
    class="emotion-0"
  />
</div>
`;
